### ISSUE TITLE
UI test fails in 2022

### ISSUE THREAD

**Issue by bmwiedemann** [bitcoin/bitcoin#13086]
When building the openSUSE bitcoin package, build fails after 2022-12-08 because the testsuite has one FAIL
in ./src/qt/test/paymentservertests.cpp that fails to verify a cert that expires in 2022 after 10y. 

would be nice to be able to build (and test) bitcoin software at any time

test failed with
```
QWARN  : PaymentServerTests::paymentServerTests() PaymentRequestPlus::getMerchant: Payment request: certificate expired or not yet active:  QSslCertificate("3", "01", "Ipbt+DxK8RDQd25/5ueXqw==", (), ("Payment Request Test Merchant"), QMap(), QDateTime(2012-12-10 16:37:24.000 UTC Qt::TimeSpec(UTC)), QDateTime(2022-12-08 16:37:24.000 UTC Qt::TimeSpec(UTC)))
FAIL!  : PaymentServerTests::paymentServerTests() Compared values are not the same
```

openSUSE has bitcoin-0.16.0

**Comment by NicolasDorier** [comment]
Ah, Bitcoin Core not supporting time traveling yet!

**Comment by DrKaradenizli** [comment]
@NicolasDorier you mean Bitcoin?

**Comment by Willtech** [comment]
May have relevance if a wallet.dat is in cold storage. Shouldn't assume that an upgrade path for the wallet will be available without building each version and upgrading it incrementally.

**Comment by laanwj** [comment]
> Shouldn't assume that an upgrade path for the wallet will be available without building each version and upgrading it incrementally unless full incremental upgrade scripts or full backward compatibility for wallet.dat is GUARANTEED in all future versions.

That sounds Kafkaesque. Have you ever noticed what nightmare it is to build old software on new platforms? 
We've always tried to keep wallet backwards compatibility as far back as possible.
(and if there is ever going to be a completely new wallet format, a strong requirement is indeed to have upgrade scripts)

**Comment by Willtech** [comment]
I thought there was a note I saw recently about not skipping versions when upgrading due to wallet upgrade - may have imagined it.

**Comment by MarcoFalke** [comment]
Fixed in 3548e4aac72e9148d41b440052a89d89328f7661
